config/lib/seed.js

```diff
@@ -2,93 +2,145 @@
-      User.generateRandomPassphrase()
-      .then(saveUser(user))
-      .catch(reportError);
-
-    } else {
-      console.log(seedAdmin.username + ' user exists');
-    }
+function removeUser (user) {
+  return new Promise(function (resolve, reject) {
+    var User = mongoose.model('User');
+    User.find({username: user.username}).remove(function (err) {
+        if (err) {
+          reject(new Error('Database Seeding:\t\t\tFailed to remove local ' + user.username));
+        }
+        resolve();
+    });
   });
-} else {
-
-  //Add Local User
-  User.find({username: seedUser.username}).remove(function () {
-    var user = new User(seedUser);
+}
 
-    // generate a random password and save
-    User.generateRandomPassphrase()
-    .then(saveUser(user))
-    .catch(reportError);
-  });
+function saveUser (user) {
+  return function() {
+    return new Promise(function (resolve, reject) {
+      // Then save the user
+      user.save(function (err, theuser) {
+        if (err) {
+          reject(new Error('Database Seeding:\t\t\tFailed to add local ' + user.username));
+        } else {
+          resolve(theuser);
+        }
+      });
+    });
+  };
+}
 
-  //Add Local Admin
-  User.find({username: seedAdmin.username}).remove(function () {
-    var user = new User(seedAdmin);
+function checkUserNotExists (user) {
+  return new Promise(function (resolve, reject) {
+    var User = mongoose.model('User');
+    User.find({username: user.username}, function (err, users) {
+        if (err) {
+          reject(new Error('Database Seeding:\t\t\tFailed to find local account ' + user.username));
+        }
 
-    // generate a random password and save
-    User.generateRandomPassphrase()
-    .then(saveUser(user))
-    .catch(reportError);
+        if (users.length === 0) {
+          resolve();
+        } else {
+          reject(new Error('Database Seeding:\t\t\tFailed due to local account already exists: ' + user.username));
+        }
+    });
   });
 }
 
+function reportSuccess (password) {
+  return function (user) {
+    return new Promise(function (resolve, reject) {
+      console.log(chalk.bold.red('Database Seeding:\t\t\tLocal ' + user.username + ' added with password set to ' + password));
+      resolve();
+    });
+  };
+}
+
 // save the specified user with the password provided from the resolved promise
-function saveUser(user) {
+function seedTheUser (user) {
   return function (password) {
-    // set the new password
-    user.password = password;
+    return new Promise(function (resolve, reject) {
+
+      var User = mongoose.model('User');
+      // set the new password
+      user.password = password;
 
-    // Then save the user
-    user.save(function (err) {
-      if (err) {
-        console.log('Database Seeding:\t\t\tFailed to add local ' + user.username);        
+      if (user.username === 'admin' && process.env.NODE_ENV === 'production') {
+        checkUserNotExists(user)
+        .then(saveUser(user))
```

Comment:
You forgot to add `reportSuccess` to the chain after successfully resolving `saveUser`.